flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontendParser.java

```diff
@@ -76,6 +76,9 @@
 	static final Option SAVEPOINT_DISPOSE_OPTION = new Option("d", "dispose", true,
 			"Disposes an existing savepoint.");
 
+	static final Option CONFIGDIR_OPTION = new Option("D", "configDir", true,
```

Comment:
I would be for removing the short option because it can easily be confused with `-d`.